### ISSUE TITLE
Allow user to disable idle timeout and display chain stats on lock screen

### ISSUE THREAD

**Issue by brandondees** [kyokan/bob-wallet#131]
Since it can take a good long while for synchronization to complete during initial setup, it would be really helpful UX wise to be able to get some idea of how things are progressing, and to help distinguish broken/stuck from just taking a while to finish. I presume it's feasible to fetch the network's current block height from hsd before it's fully caught up, and you could track sync progress relative to that, if I understand correctly.

Furthermore, it'd be nice to see synchronization status while the wallet is locked, since that seems to happen pretty quickly and isn't end-user configurable. I don't need to go to the trouble of fetching the passphrase from my vault each time if I'm just trying to check on whether sync is done yet.

**Comment by pinheadmz** [comment]
Are you talking about initial block download? There is a % sync indicator in the top right corner of the UI.

As far as Bob knowing what the sync progress is, the progress comes directly from hsd and actually given how blockchains work there is no way to really know if you are synced, but we can guess. More details about this here: https://github.com/bcoin-org/bcoin/issues/886

I suppose we could display some non-private details on top of the lock screen like node status and chain height.

**Comment by brandondees** [comment]
After setting up a new wallet just now I'm seeing the top right indicator now on the latest master. Previously (a few days ago) I couldn't get to that point, as the initial pre-wallet screen seemed to take forever with no indication of progress. 

Yeah, the lockout is a bit annoying and it'd be helpful to show some general status info from that view without having to go through the unlock procedure just to check on it. I'd also like to have the option of configuring that lock timeout or disabling it on my trusted system.

**Comment by pinheadmz** [comment]
There is some work on passphrase stuff about to be merged in for the next version:

https://github.com/kyokan/bob-wallet/pull/128

This PR is written to solve the "No Passphrase" error that comes up after a minute or so when you try to send a TX. It separates the actual wallet lock from the UI "logout".

A little background, when you enter your passphrase into hsd's wallet, it decrypts your master private key and keeps it in memory for (default) 60 seconds, then wipes it out again. This is enough time to sign a TX with the private key. My experience with Bitcoin wallets is that you always have to enter your passphrase when sending a TX (for Bob that also means opening an auction, bidding, revealing, updating, etc...) and so this PR enforces that UX.

So after #128 is merged we can address another question: Should Bob UI still lock after 5 minutes of idle time and require the passphrase to reopen the UI? In my opinion, this is a good privacy measure because your balance and transaction activity is visible on the screen.

It sounds like your preference would be to either remove the idle logout altogether, or display some interesting network stats on the lock screen itself so you can at least follow some activity and know your node is always in sync (could also indicate # of peers, etc.)

I think maybe we could just allow the user to disable the idle logout, or change the amount of time.

Finally, https://github.com/kyokan/bob-wallet/pull/117 also introduces a new "wallet sync progress" modal, so when you import a new seed on an already-synced Bob, the wallet progress is displayed and blocks the UI until finished (was this your question in the first place? -- there is the blockchain sync and the wallet sync, they're separate)

**Comment by brandondees** [comment]
I think all the security and privacy safeguards should definitely be _available_ and even _on by default_ but as a technical power user with a well hardened system that is also pretty well physically secured, I am comfortable with leaving software "unlocked" while it's running for the sake of drastically improved ease of use (my passphrases used for cryptocurrency stuff are not practically type-able so it's a whole ceremony to retrieve and copy/paste it each time). I'd just like to have the option, with a scary warning, to disable the auto-locking since I know what I'm doing.

**Comment by brandondees** [comment]
Regardless of the above, it'd definitely still be nice for me to see whatever non-sensitive info is reasonable to see from the lock screen, like sync state, block height, connectivity, whatever else.

**Comment by brandondees** [comment]
To answer the last question, I think this app just doesn't work reliably yet on linux and each time I build it and set up my wallet, it seems to go smoothly and sync up within a matter of minutes. After I close the app and re-open it, it hangs on the initial "Loading node..." screen and never seems to progress past that.

I can delete my `~/.config/Bob/` contents and get the first-run experience again but sometimes it takes longer and it's hard to tell what's up.

**Comment by mackuba** [comment]
+1 for configurable auto-logout time, it's very annoying…

**Comment by rithvikvibhu** [comment]
Customizable timeouts and displaying sync status on lock screen has been added since v0.8.0, closing this issue. Feel free to re-open if there's more to discuss (or maybe start a new issue).
